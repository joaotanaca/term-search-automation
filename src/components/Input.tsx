import styled from "styled-components";

export const Input = styled.input`
    background: #eaedf3;
    border-radius: 8px;
`;
