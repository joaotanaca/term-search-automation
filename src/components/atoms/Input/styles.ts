import styled, { css } from "styled-components";

export const Container = styled.div<{ error?: string }>`
    font-weight: 400;
    display: inline-flex;
    flex-direction: column;
    gap: 0.5em;
    label {
        color: #2d74ff;
        font-size: 0.85em;
        font-weight: inherit;
    }
    input {
        padding: 10px 0 10px 12px;
        border: none;
        outline: none;
        background: #eaedf3;
        border-radius: 0.5em;
        color: #364663;
        font-size: 1em;
        font-weight: inherit;
        &:disabled {
            background-color: #eaedf3;
            color: #96a6c2;
        }
    }

    ${({ error, "aria-disabled": ariaDisabled }) =>
        error &&
        !ariaDisabled &&
        css`
            label {
                color: #ff2d2d;
            }
            input {
                border: 1px solid #ff2d2d;
                background-color: rgba(255, 45, 45, 0.1);
                color: #ff2d2d;
            }
            ::after {
                content: "${error}";
                font-size: 0.85em;
                color: #364663;
                font-weight: inherit;
            }
        `}
`;
