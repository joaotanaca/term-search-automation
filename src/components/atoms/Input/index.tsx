import React, { PropsWithChildren, useMemo } from "react";
import { Container } from "./styles";

type Props = React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
> & { label?: string; error?: string };

const Input = ({
    label,
    error,
    disabled,
    ...props
}: PropsWithChildren<Props>) => {
    const renderLabel = useMemo(
        () =>
            label && !disabled ? (
                <label htmlFor={props.name}>{label}</label>
            ) : null,
        [disabled],
    );
    return (
        <Container aria-disabled={disabled} error={error}>
            {renderLabel}
            <input disabled={disabled} {...props} />
        </Container>
    );
};

export default Input;
