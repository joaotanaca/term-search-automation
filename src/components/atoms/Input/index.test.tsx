import { render, fireEvent, renderer } from "@test";
import Input from ".";

const ARIAL_LABEL = "test-label";
const NAME = "test-input";
const ERROR = "ERROR DESCRIPTION";

it("must validate if the value changes", () => {
    const container = render(<Input aria-label={ARIAL_LABEL} />);
    const input = container.getByLabelText(ARIAL_LABEL) as HTMLInputElement;

    expect(input.value).toEqual("");

    fireEvent.change(input, { target: { value: "23" } });

    expect(input.value).toEqual("23");
});

it("must validate if the name attribute is being correctly assigned", () => {
    const container = render(<Input name={NAME} aria-label={ARIAL_LABEL} />);
    const input = container.getByLabelText(ARIAL_LABEL) as HTMLInputElement;

    expect(input.name).toEqual(NAME);
});

it("must validate if the disabled attribute is being correctly assigned", () => {
    const container = render(
        <Input disabled={true} aria-label={ARIAL_LABEL} />
    );
    const input = container.getByLabelText(ARIAL_LABEL) as HTMLInputElement;

    expect(input.disabled).toBeTruthy();

    fireEvent.change(input, { target: { disabled: false } });

    expect(input.disabled).toBeFalsy();
});

it("validate if the snapshot has the error styling", async () => {
    const container = renderer
        .create(<Input error={ERROR} aria-label={ARIAL_LABEL} />)
        .toJSON();

    expect(container).toMatchSnapshot();
});
