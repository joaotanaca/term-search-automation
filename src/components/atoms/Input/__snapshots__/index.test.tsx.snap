// Vitest Snapshot v1

exports[`validate if the snapshot has the error styling 1`] = `
.c0 {
  font-weight: 400;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5em;
}

.c0 label {
  color: #2d74ff;
  font-size: 0.85em;
  font-weight: inherit;
}

.c0 input {
  padding: 10px 0 10px 12px;
  border: none;
  outline: none;
  background: #eaedf3;
  border-radius: 0.5em;
  color: #364663;
  font-size: 1em;
  font-weight: inherit;
}

.c0 input:disabled {
  background-color: #eaedf3;
  color: #96a6c2;
}

.c0 label {
  color: #ff2d2d;
}

.c0 input {
  border: 1px solid #ff2d2d;
  background-color: rgba(255,45,45,0.1);
  color: #ff2d2d;
}

.c0::after {
  content: "ERROR DESCRIPTION";
  font-size: 0.85em;
  color: #364663;
  font-weight: inherit;
}

<div
  className="c0 input-container"
>
  <input
    aria-label="test-label"
  />
</div>
`;
